### PR TITLE
Dark Mode

### PR DESCRIPTION
**Pull request by mehmetsagir**
I was wondering how the Animate.style website would look in dark mode and I tried it. I liked it and wanted to share it with you. You can view the renewed website here.

- [animate.style](https://animatecss-ms.netlify.app/)
- Additionally, I added a back to top button.

### CODE REVIEW

**Comment by WarenGonzaga** [comment]
@mehmetsagir thanks... I'll review this and I would like to improve your progress. Thank you for taking time to work on this as I'm bit busy for this update.

**Comment by WarenGonzaga** [comment]
@mehmetsagir kindly fix the remaining conflicting issue on your PR . thank you

**Comment by mehmetsagir** [comment]
@WarenGonzaga okay, conflicts fixed

File: docs/index.html [hunk 1]
```diff
@@ -51,6 +51,20 @@ <h2 class="callout-subtitle">Just-add-water CSS animations</h2>
 
       <section class="animation-list">
         <button class="callout-hideList button">Close list</button>
+
+        <a class="dark-mode-button-container">
```
Comment (eltonmesquita) [inline docs/index.html:55]: You have to make these changes in the [`docsSource/template.html`](https://github.com/animate-css/animate.css/blob/main/docsSource/template.html) file.

You'll also need to compile the docs using `npm run docs`.

File: docs/main.mjs [hunk 2]
```diff
@@ -23,3 +29,37 @@ document.querySelectorAll('.copy-icon').forEach(icon => {
         }, 750)
     })
 })
+
+backToTopBtn.addEventListener('click', (e) => {
```
Comment (eltonmesquita) [inline docs/main.mjs:33]: Let's avoid adding lot's of things to this file. Create a new module file in the [modules](https://github.com/animate-css/animate.css/tree/main/docs/modules) folder and wrap it all in an exported function which will get imported and executed in this file. See the other files in this folder for examples.

File: docs/main.mjs [hunk 3]
```diff
@@ -23,3 +29,37 @@ document.querySelectorAll('.copy-icon').forEach(icon => {
...
+    window.scrollTo(0, 0);
+})
+
+darkCheck.addEventListener('click', () => {
+    if(darkCheck.checked){
+        document.body.className = 'dark';
+        localStorage.setItem('animate-css', 'dark');
+        moon.style.fill = '#e0e0e0';
+        sun.style.fill = '#e0e0e0';
+    }else{
```
Comment (eltonmesquita) [inline docs/main.mjs:44]: please, format your code properly: `} else {`.

File: docs/main.mjs [hunk 4]
```diff
@@ -23,3 +29,37 @@ document.querySelectorAll('.copy-icon').forEach(icon => {
...
+        document.body.className = '';
+        localStorage.removeItem('animate-css');
+        moon.style.fill = '#111';
+        sun.style.fill = '#111';
+    }
+})
+
+window.addEventListener("scroll", () => {
```
Comment (eltonmesquita) [inline docs/main.mjs:52]: please add `{ passive: true }` as a third argument to the `addEventListener` function. [See more about it here](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#improving_scrolling_performance_with_passive_listeners).

File: docs/main.mjs [hunk 5]
```diff
@@ -23,3 +29,37 @@ document.querySelectorAll('.copy-icon').forEach(icon => {
...
+  if (window.pageYOffset > 400) {
+    backToTopBtn.style.display = "grid";
```
Comment (eltonmesquita) [inline docs/main.mjs:54]: Please switch all your double quotes `"` for single ones `'`.

File: docs/style.css [hunk 6]
```diff
@@ -616,3 +616,126 @@ blockquote {
 .animation-list {
   text-align: left;
 }
+
+/*-----------------------------------*\
+  $DARK-MODE
+\*-----------------------------------*/
+
+.night-light-toggle-button .night-light-checkbox {
```
Comment (eltonmesquita) [inline docs/style.css:624]: please avoid descended selector, it complicates your CSS and couple HTML to CSS too much.

File: docs/style.css [hunk 7]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+  position: absolute;
+  visibility: hidden;
+}
```
Comment (eltonmesquita) [inline docs/style.css:627]: Please, add a new line between declarations.

File: docs/style.css [hunk 8]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+.night-light-checkbox:checked + .night-light-label .night-light-ball {
```
Comment (eltonmesquita) [inline docs/style.css:628]: Do we need descendent selector here?

File: docs/style.css [hunk 9]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+  transform: translateX(28px);
+}
```
Comment (eltonmesquita) [inline docs/style.css:630]: new line

File: docs/style.css [hunk 10]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+  background-color: #4672fe;
+  border-radius: 50%;
+  transition: transform 0.3s linear;
+}
+.moon-svg,
+.sun-svg {
+  width: 17px;
+}
+
+body.dark{
```
Comment (eltonmesquita) [inline docs/style.css:658]: remove the tag selector `body` as it's unnecessary.

File: docs/style.css [hunk 11]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+  background-color: #111;
+  color: #e0e0e0;
+}
+
+body.dark .intro{
```
Comment (eltonmesquita) [inline docs/style.css:663]: remove `body`

File: docs/style.css [hunk 12]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+  background: #111;
+}
+
+body.dark .animation-list,
```
Comment (eltonmesquita) [inline docs/style.css:667]: remove `body`

File: docs/style.css [hunk 13]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+body.dark .sidebar{
```
Comment (eltonmesquita) [inline docs/style.css:668]: remove `body`

File: docs/style.css [hunk 14]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+  background: #15151D;
+}
+
+body.dark .button.button-animations,
```
Comment (eltonmesquita) [inline docs/style.css:672]: remove `body`

File: docs/style.css [hunk 15]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+body.dark .callout-hideList.button{
```
Comment (eltonmesquita) [inline docs/style.css:673]: remove `body`

File: docs/style.css [hunk 16]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+  color: #e0e0e0;
+  border-color: rgba(255, 255, 255, .7);
+  outline: none;
+}
+
+body.dark .copy-icon,
```
Comment (eltonmesquita) [inline docs/style.css:679]: remove `body`

File: docs/style.css [hunk 17]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+body.dark .copy-icon:hover{
```
Comment (eltonmesquita) [inline docs/style.css:680]: remove `body`

File: docs/style.css [hunk 18]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+  place-content: center;
+  cursor: pointer;
+  z-index: 999;
+}
+
+.back-to-top .up-svg{
+  width: 13px;
+}
+
+.back-to-top .up-svg path{
```
Comment (eltonmesquita) [inline docs/style.css:739]: `.back-to-top` is unnecessary, please remove it. Also, format your code properly. Always add a space before a `{`

File: docs/style.css [hunk 19]
```diff
@@ -616,3 +616,126 @@ blockquote {
...
+  color: #111;
+}
+
+body.dark h2,
+body.dark h3,
+body.dark h4,
+body.dark a[title="Documentation"],
+body.dark pre *{
+  color: #e0e0e0;
+}
```
Comment (eltonmesquita) [inline docs/style.css:700]: new line

**Review by eltonmesquita — CHANGES_REQUESTED** [review]
Sorry for the late review, but you know, RealLife™️. Thanks for the PR it's looking great!

I added a bunch of comments, but they should be simple to fix. Most of them are code standards related as we currently don't have one 😅,  I'll try to work on that later. Don't feel down for having so many comments when PRing in open source projects, this is fairly usual.

I noticed that you have a bunch of commits and they have unrelated titles. Can you try to fix that? You can ask for help if you need, but I think it'd be easier if you use a GUI like SourceTree to undo the commits and properly name them as new commits. You'll have to force push your work, but it shouldn't be a problem.

**Comment by eltonmesquita** [comment]
If you need any help, feel free to ask. I might take a bit long but I'll gladly help!

File: docs/style.css (hunk 18 above)
Comment (WarenGonzaga) [reply docs/style.css:739]: Good catch!

**Review by WarenGonzaga — COMMENTED** [review]
Looks good to me... can you provide a preview of it?

**Comment by mehmetsagir** [comment]
Elbette, [demo-animate.style](https://animatecss-ms.netlify.app/) üzerinden inceleyebilirsiniz. If there are places I need to fix again, just specify.

**Comment by WarenGonzaga** [comment]
Hi @mehmetsagir can you fit this slider button?
image

Looks like not fit and I saw spacing...

**Comment by WarenGonzaga** [comment]
A good reference might like this...
image

**Comment by mehmetsagir** [comment]
Ok i will edit

**Comment by WarenGonzaga** [comment]
Awesome @mehmetsagir let us know... and don't forget to prepare an update preview.

**Comment by mehmetsagir** [comment]
I edited the dark mode button. I deleted the blanks and simplified the code a bit. 
@WarenGonzaga, you can view domo-animate.css here. I would appreciate it if you let me know if there are places I need to organize.

**Review by WarenGonzaga — APPROVED** [review]
Smooth! Nice update! @eltonmesquita kindly review this... for me its all good.

File: docs/modules/darkMode.mjs [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+  } else {
+    document.body.classList.remove('dark');
+    localStorage.removeItem('animate-css');
+  }
+})
+
+if (localStorage.getItem('animate-css')) {
+  document.body.className = 'dark';
+  darkCheck.checked = true;
+}
```
Comment (eltonmesquita) [inline docs/modules/darkMode.mjs:16]: Could you please wrap this whole file in a function and export it just like the other modules?

File: docs/main.mjs [hunk 21]
```diff
@@ -2,6 +2,7 @@ import buildDocsIndex from './modules/buildDocsIndex.mjs';
 import playground from './modules/playground.mjs';
 import startAnimations from './modules/startAnimations.mjs';
 import toggleOnClick from './modules/toggle.mjs';
+import('./modules/darkMode.mjs');
```
Comment (eltonmesquita) [inline docs/main.mjs:5]: Please use a named import and execute the function bellow.

**Review by eltonmesquita — CHANGES_REQUESTED** [review]
Thanks a lot for all the effort, it's looking great! One thing that is not yet addressed is the git commit history. It's just too messy right now and I prefer to avoid merging it as it is right now. In my opinion, one single commit would be enough for this PR. Make sure to also give a short, yet descriptive, commit message.

It can be daunting to rewrite git history, but it is easier than it seems. [This article](https://www.atlassian.com/git/tutorials/rewriting-history) is a good intro into git history rewriting. You can also use a GUI as I mentioned before. Feel free to ask for help if you're having a hard time doing this.

**Comment by mehmetsagir** [comment]
I fixed the commit messages of all the pages I was editing. I'm not sure if it's correct, can you review when appropriate? @eltonmesquita

**Review by eltonmesquita — APPROVED** [review]
Looks good! Thanks for all the effort 🎖️

**Pull request merged**